lib/WUI/nhttp/gcode_upload.cpp

```diff
@@ -222,22 +222,67 @@ UploadHooks::Result GcodeUpload::data(std::string_view data) {
 
 namespace {
 
-    UploadHooks::Result prepend_usb_path(const char *filename, char *filepath_out, size_t filepath_size) {
-        const uint32_t fname_length = strlen(filename);
-
-        if ((fname_length + USB_MOUNT_POINT_LENGTH) >= filepath_size) {
+    UploadHooks::Result prepend_usb_path(string_view filename, char *filepath_out, size_t filepath_size) {
+        if ((filename.length() + USB_MOUNT_POINT_LENGTH) >= filepath_size) {
             // The Request header fields too large is a bit of a stretch...
             return make_tuple(Status::RequestHeaderFieldsTooLarge, "File name too long");
         }
         strlcpy(filepath_out, USB_MOUNT_POINT, USB_MOUNT_POINT_LENGTH + 1);
-        strlcat(filepath_out, filename, FILE_PATH_BUFFER_LEN - USB_MOUNT_POINT_LENGTH);
+        strncat(filepath_out, filename.data(), filename.length());
+        filepath_out[USB_MOUNT_POINT_LENGTH + filename.length()] = '\0';
+
+        return make_tuple(Status::Ok, nullptr);
+    }
+
+    UploadHooks::Result make_dir(string_view dir) {
+        char path[USB_MOUNT_POINT_LENGTH + dir.length() + 1];
+
+        auto error = prepend_usb_path(dir, path, sizeof(path));
+        if (std::get<0>(error) != Status::Ok)
+            return error;
+        if (mkdir(path, 777) != 0) {
```

Comment:
Again, braces… (below too).

Reply:
Also… what if the folder already exists. Is that OK to create it „again“?

Reply:
AFAIK it is, it just fails with `errno == EEXIST` for which I check.